src/test/frameworks/compose/discover/frameworkMatcher.spec.ts

```diff
@@ -0,0 +1,168 @@
+      const allFrameworks: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [],
+        },
+        {
+          id: "next",
+          runtime: "nodejs",
+          requiredDependencies: [],
+          embedsFrameworks: ["react"],
+        },
+        {
+          id: "react",
+          runtime: "nodejs",
+          requiredDependencies: [],
+        },
+      ];
+      const actual = removeEmbededFrameworks(allFrameworks);
+      const expected: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [],
+        },
+        {
+          id: "next",
+          runtime: "nodejs",
+          requiredDependencies: [],
+          embedsFrameworks: ["react"],
+        },
+      ];
+
+      expect(actual).to.have.deep.members(expected);
+    });
+  });
+
+  describe("filterFrameworksWithFiles", () => {
+    it("should return frameworks having all the required files", async () => {
+      const allFrameworks: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [],
+          requiredFiles: [["package.json", "package-lock.json"]],
+        },
+        {
+          id: "next",
+          runtime: "nodejs",
+          requiredDependencies: [],
+          requiredFiles: [["next.config.js"], "next.config.ts"],
+        },
+      ];
+      const actual = await filterFrameworksWithFiles(allFrameworks, fileSystem);
+      const expected: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [],
+          requiredFiles: [["package.json", "package-lock.json"]],
+        },
+      ];
+
+      expect(actual).to.have.deep.members(expected);
+    });
+  });
+
+  describe("filterFrameworksWithDependencies", () => {
+    it("should return frameworks having required dependencies with in the project dependencies", () => {
+      const allFrameworks: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [
+            {
+              name: "express",
+            },
+          ],
+        },
+        {
+          id: "next",
+          runtime: "nodejs",
+          requiredDependencies: [
+            {
+              name: "next",
+            },
+          ],
+        },
+      ];
+      const projectDependencies: Record<string, string> = {
+        express: "^4.18.2",
+      };
+      const actual = filterFrameworksWithDependencies(allFrameworks, projectDependencies);
+      const expected: FrameworkSpec[] = [
+        {
+          id: "express",
+          runtime: "nodejs",
+          requiredDependencies: [
+            {
+              name: "express",
+            },
+          ],
+        },
+      ];
+
+      expect(actual).to.have.deep.members(expected);
```

Comment:
I think you should also check to make sure `actual` doesn't contain any other members.